flanker/mime/message/part.py

```diff
@@ -665,18 +665,32 @@ def stronger_encoding(a, b):
 
 
 def has_long_lines(text, max_line_len=599):
-    '''
+    """
     Returns True if text contains lines longer than a certain length.
     Some SMTP servers (Exchange) refuse to accept messages "wider" than
     certain length.
-    '''
+    """
     if not text:
         return False
     for line in text.splitlines():
         if len(line) >= max_line_len:
             return True
     return False
 
+def _base64_decode(s):
+    """Recover base64 if it is broken."""
+    try:
+        return base64.b64decode(s)
+    except TypeError:
+        s = s.replace("\n", "").replace("\r", "")
```

Comment:
Turns out `replace` does not allocate if nothing needs to be replace. But still you scan O(n) the string twice.